[pt2] Fix multiple races in log folder

xw285cornell opened the pull request:
Summary:
There are a few races/permission errors in file creation, fixing
OSS:
1. caffe2/torch/_dynamo/utils.py, get_debug_dir: multiple process may conflict on it even it's using us. Adding pid to it
2. caffe2/torch/_dynamo/config.py: may not be a right assumption that we have permission to cwd

Test Plan: sandcastle

Differential Revision: D42905908



cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire @chunyuan-w

mlazos reviewed (changes requested):
Rather than changing the config default for the debug dir location for everyone, please patch this for internal code

xw285cornell commented on torch/_dynamo/config.py:
```diff
@@ -181,7 +182,7 @@
 base_dir = dirname(dirname(dirname(abspath(__file__))))
 
-debug_dir_root = os.path.join(os.getcwd(), "torch_compile_debug")
+debug_dir_root = os.path.join(tempfile.gettempdir(), "torch_compile_debug")
```
I think there maybe a few things: (1) often times cwd is under source control so it'll add files that can make git/hg a little messy; (2) sometimes we don't have permission on cwd; also it's a little weird(?) that if we run the binary under different folder it'll end up different place; change to /tmp should get rid of those problems; (3) the logic here won't work as os.rename doesn't work across different file systems https://github.com/pytorch/pytorch/blob/d690a596dcbccf943f90c3e5353394cadc29dfd1/torch/_inductor/debug.py#L267

mlazos reviewed (approved):
Thanks!!

xw285cornell committed:
[pt2] Fix multiple races in log folder (#93407)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/93407

There are a few races/permission errors in file creation, fixing
OSS:
1. caffe2/torch/_dynamo/utils.py, get_debug_dir: multiple process may conflict on it even it's using us. Adding pid to it
2. caffe2/torch/_dynamo/config.py: may not be a right assumption that we have permission to cwd

Test Plan: sandcastle

Reviewed By: soumith, aaronenyeshi, brad-mengchi, jiawenliu64

Differential Revision: D42905908

fbshipit-source-id: 395814953af36ded78229c4a83ced66561bccfb1
